go/ql/lib/semmle/go/security/HardcodedCredentials.qll

```diff
@@ -0,0 +1,158 @@
+/**
+ * Provides default sources, sinks and sanitizers for reasoning about
+ * hardcoded credentials, as well as extension points
+ * for adding your own.
+ */
+
+import go
+private import semmle.go.StringOps
+private import semmle.go.dataflow.ExternalFlow
+private import semmle.go.security.Jwt
+
+/**
+ * Provides default sources, sinks and sanitizers for reasoning about
+ * hardcoded credentials, as well as extension points
+ * for adding your own.
+ */
+module HardcodedCredentials {
+  /** A data flow source for hardcoded credentials. */
+  abstract class Source extends DataFlow::Node { }
+
+  /** A data flow sink for hardcoded credentials. */
+  abstract class Sink extends DataFlow::Node { }
+
+  /** A sanitizer for hardcoded credentials. */
+  abstract class Sanitizer extends DataFlow::Node { }
+
+  private module Config implements DataFlow::ConfigSig {
+    predicate isSource(DataFlow::Node source) { source instanceof Source }
+
+    predicate isSink(DataFlow::Node sink) { sink instanceof Sink }
+
+    predicate isBarrier(DataFlow::Node node) { node instanceof Sanitizer }
+  }
+
+  /** Tracks taint flow for reasoning about hardcoded credentials. */
+  module Flow = TaintTracking::Global<Config>;
+
+  /** A hardcoded string literal as a source for hardcoded credentials. */
+  private class HardcodedStringSource extends Source {
+    HardcodedStringSource() { this.asExpr() instanceof StringLit }
```

Comment:
I note a few other queries / languages have some exclusions for obviously non-credential strings (empty string, password123, etc)-- as a followup, worth considering adopting similar here.

Reply:
Good point as well! Done in https://github.com/github/codeql/pull/15527/commits/2a3dfbe99f254d0ed42f10c6eb19f05f8c3db71a.